specs/templates/table-column-hld.md

```diff
@@ -0,0 +1,133 @@
+
+## Implementation / Design
+
+*Describe interesting implementation/design decisions, with justifications and alternates considered.*
+
+*Include code snippets showing basic component use and any interesting configurations.*
+
+Typical usage:
+
+```html
+<nimble-table>
+    <nimble-table-column-*>
+        Column Label
+    </nimble-table-column-*>
+</nimble-table>
+```
+
+### API
+
+#### Column Element
+
+_Element Name_
+
+-   `nimble-table-column-*`
+
+_Props/Attrs_
+
+-   `attr-name`: string (defaults to ...) - Description of the property
+
+_Content_
+
+-   column label (icon and/or text)
+
+_Events_
+
+*List any events along with event detail interface(s).*
+
+#### Cell View Element
+
+_Element Name_
+
+-   `nimble-table-column-*-cell-view`
+
+_Rendering_
+
+*Describe what the cell view will render.*
+
+#### Group Header View Element
+
+*If an existing group header view element will be used instead of creating a new one:*
+
+    A new element will not be created for the group header view. The column will specify the existing `tableColumnTextGroupHeaderViewTag` as the `groupHeaderViewTag` because the header will contain only text.
+
+*Otherwise:*
+
+_Element Name_
+
+-   `nimble-table-column-*-group-header-view`
+
+_Rendering_
+
+*Describe what the group header view will render.*
+
+### Sorting / Grouping
+
+*Will the column be sortable and/or groupable? If so, which `xxx-field-name` attribute will be assigned to  `operandDataRecordFieldName`? If sortable, which `TableColumnSortOperation` will be used? Describe any interesting use cases.*
+
+
+### Sizing
+
+*Will the column use a fractional width, fixed width, or both (via `width-mode` attribute)?*
+
+### Placeholder
+
+*What is the behavior of cells and group rows when the value is `undefined` or `null`? Will a `placeholder` be supported to display a specific interpretation for missing values (e.g. showing "Never run" in a date-time column of latest execution times)? Are there additional values (e.g. empty string) for which placeholders or default rendering need to be considered?*
+
+### Delegated Events
+
+*Are there any events emitted by components in cells that a client might be interested in? Because clients cannot easily subscribe to every cell view, such events should be delegated to the column element so that it can emit an equivalent event for clients to subscribe to. Names of delegated events (e.g. `'click'`, `'keydown'`, etc.) should be listed in the `delegatedEvents` property of the column internals options, and the column should set up a listener to handle the `delegated-event` event that emits a new event for clients to listen for. For example, `click` events from the cell might be delegated to the column and re-emitted as `button-click` events. Describe here any events that will be delegated (e.g. `click`), the new event name(s) that will be emitted by the column (e.g. `button-click`), and any extra data that will be included in those event details (e.g. `recordId` field).*
```

Comment:
this is pretty lengthy, maybe some line breaks

```
*Are there any events emitted by components in cells that a client might be interested in? Because clients cannot easily subscribe to every cell view, such events should be delegated to the column element so that it can emit an equivalent event for clients to subscribe to. Names of delegated events (e.g. `'click'`, `'keydown'`, etc.) should be listed in the `delegatedEvents` property of the column internals options, and the column should set up a listener to handle the `delegated-event` event that emits a new event for clients to listen for.*

*For example, `click` events from the cell might be delegated to the column and re-emitted as `button-click` events. Describe here any events that will be delegated (e.g. `click`), the new event name(s) that will be emitted by the column (e.g. `button-click`), and any extra data that will be included in those event details (e.g. `recordId` field).*
```

Think it's also helpful to just have an example folks can edit instead of describing what an example would look like, maybe something like:
```
- delegated event name: `click` (name of event captured from the cell-view to delegate to the column)
- re-emitted event name: `button-column-click` (name of event emitted from the column)
- re-emitted event detail: `recordId` (extra details included)
```

Reply:
I gave an example written in sentences, as opposed to just a list.